### ISSUE TITLE
onEmbeddedFactoryInit NullPointerException

### ISSUE THREAD

**Issue by realkarmakun** [orientechnologies/orientdb#9629]
## Expected behavior  
I expect embedded OrientDB instance to startup.

## Actual behavior  
It is throwing NPE at the start
```
2021-06-28 15:24:01:792 INFO  Loading configuration from input stream [OServerConfigurationLoaderXml]
2021-06-28 15:24:02:184 INFO  OrientDB Server v3.1.12 - Veloce (build b225e767a05a41efa9279a0084ade3fdcec8ab92, branch UNKNOWN) is starting up... 
[OServer]java.lang.NullPointerException
	at com.orientechnologies.orient.core.Orient.onEmbeddedFactoryInit(Orient.java:974)
	at com.orientechnologies.orient.core.db.OrientDBEmbedded.<init>(OrientDBEmbedded.java:107)
	at com.orientechnologies.orient.core.db.OrientDBInternal.embedded(OrientDBInternal.java:110)
	at com.orientechnologies.orient.server.OServer.startupFromConfiguration(OServer.java:401)
	at com.orientechnologies.orient.server.OServer.startup(OServer.java:335)
	at ru.meproject.rustmesponge.RustMeSponge.onServerStarting(RustMeSponge.java:52)
	at org.spongepowered.common.event.listener.StartingEngineEventListener_RustMeSponge_onServerStarting1.handle(Unknown Source)
	at org.spongepowered.vanilla.launch.event.RegisteredListener.handle(RegisteredListener.java:97)
	at org.spongepowered.vanilla.launch.event.VanillaEventManager.post(VanillaEventManager.java:418)
	at org.spongepowered.vanilla.launch.event.VanillaEventManager.post(VanillaEventManager.java:448)
	at org.spongepowered.common.SpongeLifecycle.callStartingEngineEvent(SpongeLifecycle.java:186)
	at net.minecraft.server.dedicated.DedicatedServer.handler$znh000$vanilla$runEngineStartLifecycle(SourceFile:1204)
	at net.minecraft.server.dedicated.DedicatedServer.initServer(SourceFile)
	at net.minecraft.server.MinecraftServer.runServer(SourceFile:645)
	at net.minecraft.server.MinecraftServer.lambda$spin$0(SourceFile:257)
	at java.lang.Thread.run(Thread.java:748)
```

## Steps to reproduce  
In my app I have startup events, on which I'm creating OrientDB instance like this:
```java
        try {
            logger.info("Starting OrientDB");
            orientDb = new OServer(true);
            orientDb.startup(getClass().getResourceAsStream("/orientdb.xml"));
        } catch (Exception e) {
            logger.error("Something gone wrong:");
            e.printStackTrace();
        }
```

And shutdown event:
```java
        logger.info("Shutting down OrientDB");
        orientDb.shutdown();
```

my orientdb.xml file:
```xml
<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<orient-server>
   <properties>
      <entry value="mods/test.db" name="server.database.path"/>
      <entry name="cache.size" value="10000" />
      <entry name="storage.keepOpen" value="true" />
      <entry name="server.cache.staticResources" value="false"/>
      <entry name="log.console.level" value="info"/>
      <entry name="log.file.level" value="fine"/>
      <entry name="plugin.dynamic" value="false"/>
   </properties>
</orient-server>
```

If you wondering, I'm making a SpongeAPI plugin for Minecraft server. I think error is not related to this, since HikariPool starts and shuts down just fine.

**Comment by realkarmakun** [comment]
Also here is my buildscript:
```kotlin
dependencies {
    implementation("com.orientechnologies:orientdb-core:3.1.12")
    implementation("com.orientechnologies:orientdb-client:3.1.12")
    implementation("com.orientechnologies:orientdb-server:3.1.12")
    implementation("com.orientechnologies:orientdb-tools:3.1.12")
    implementation("com.sun.jna:jna:3.0.9")
    implementation("com.googlecode.concurrentlinkedhashmap:concurrentlinkedhashmap-lru:1.4.2")
    implementation("org.lz4:lz4-java:1.8.0")
}
```
I think I'm missing out on some kind of configuration, but I've gone through docs on Embedded server several times now, and still have no idea what is wrong

**Comment by skywalker123p** [comment]
Is there any solution to this problem? I have also same issue. Thanks for reply.

**Comment by tglman** [comment]
Hi,

This should be fixed in 3.2.17, can you double check it ?

Regards
